src/main/java/emu/grasscutter/game/managers/InsectCaptureManager.java

```diff
@@ -0,0 +1,73 @@
+package emu.grasscutter.game.managers;
+
+import java.util.HashMap;
+
+import emu.grasscutter.Grasscutter;
+import emu.grasscutter.data.GameData;
+import emu.grasscutter.data.excels.ItemData;
+import emu.grasscutter.game.entity.EntityMonster;
+import emu.grasscutter.game.inventory.GameItem;
+import emu.grasscutter.game.player.Player;
+import emu.grasscutter.game.props.ActionReason;
+import emu.grasscutter.net.proto.VisionTypeOuterClass;
+
+public class InsectCaptureManager {
+    private final Player player;
+    //Monster ID -> Item ID
+    private static final HashMap<Integer,Integer> insectsMapping = new HashMap<>(); //HashMap<Monster Id,Item Id>
+    public InsectCaptureManager(Player player) {
+        this.player = player;
+    }
+    public void arrestSmallCreature(EntityMonster monster){
+        int monsterId = monster.getMonsterData().getId();
+        if(insectsMapping.containsKey(monsterId)) {
+            int itemId = insectsMapping.get(monsterId);
+            ItemData data = GameData.getItemDataMap().get(itemId);
+            GameItem item = new GameItem(data, 1);
+            player.getInventory().addItem(item, ActionReason.SubfieldDrop);
+            monster.getScene().removeEntity(monster, VisionTypeOuterClass.VisionType.VISION_TYPE_REMOVE);
+        }else{
+            Grasscutter.getLogger().warn("Monster(id={}) couldn't be caught. please EDIT InsectCaptureManager class to allow it", monsterId);
+        }
+    }
+    static{
+        //Monsters(Id) who can be caught to Item Id

```

Comment:
This shouldn't be hardcoded. I'll dig around later and see if this can be derived from the bindata. If it can't, it should be a `.json` config file.

Reply:
Ok I think this file has the data
https://github.com/Dimbreath/GenshinData/blob/ebb117f78dab56e704853b71fa60f45ee2cefe79/ExcelBinOutput/EnvAnimalGatherExcelConfigData.json#L19-L28

Reply:
> Ok I think this file has the data https://github.com/Dimbreath/GenshinData/blob/ebb117f78dab56e704853b71fa60f45ee2cefe79/ExcelBinOutput/EnvAnimalGatherExcelConfigData.json#L19-L28

Are you leisure to edit this code with your suggestion? I have to take a class unti afternoon(China Time),and being not familiar with reading json file data, I have to read other code to learn about it ,thank you very much.

Reply:
I'm a bit busy today too, someone else may be able to link an example of how to read from bindata in the meantime.

Reply:
> I'm a bit busy today too, someone else may be able to link an example of how to read from bindata in the meantime.

ok,being leisure after class, I will refer other's code to refine the code

Reply:
From @Melledy:
>from binoutput files, you should create class files for them and load them from the resource loader
>excel files are auto loaded from any GameResource class (you just need to provide the schema for them)
>you can find examples of excel data in grasscutter.data.excels i believe